src/utils/serve-functions.js

```diff
@@ -305,10 +305,12 @@ const createFormSubmissionHandler = function ({ siteUrl }) {
           })
         })
       } catch (error) {
-        return console.error(error)
+        warn(error)
```

Comment:
Both these error cases didn't call `next` so the server hanged instead of returning `404`